RevenueCatUI/Templates/V2/Components/Carousel/CarouselComponentView.swift

```diff
@@ -0,0 +1,693 @@
+
+        if loop {
+            // Start with 3 copies so user can swipe freely left or right.
+            let firstCopy  = makeItems(forCopyIndex: 0)
+            let secondCopy = makeItems(forCopyIndex: 1)
+            let thirdCopy  = makeItems(forCopyIndex: 2)
+
+            data = firstCopy + secondCopy + thirdCopy
+
+            // Put user in the middle copy
+            index = originalCount + self.initialIndex
+        } else {
+            // Non-looping: just one copy
+            data = makeItems(forCopyIndex: 0)
+            index = self.initialIndex
+        }
+    }
+
+    /// Create one “copy” of the original pages, each with a unique ID
+    /// so SwiftUI knows these are distinct items from other copies.
+    private func makeItems(forCopyIndex copyIndex: Int) -> [CarouselItem<Content>] {
+        originalPages.enumerated().map { (pageIndex, view) in
+            let uniqueID = copyIndex * originalCount + pageIndex
+            return CarouselItem(id: uniqueID, view: view)
+        }
+    }
+
+    // MARK: - Auto-Play
+
+    private func startAutoPlayIfNeeded() {
+        guard let msTimePerSlide = msTimePerSlide,
+              let msTransitionTime = msTransitionTime else { return }
+
+        autoTimer?.invalidate() // Stop any existing timer
+
+        autoTimer = Timer.scheduledTimer(withTimeInterval: Double(msTimePerSlide) / 1000, repeats: true) { _ in
+            guard !isPaused else {
+                // If paused, check if 10 seconds have passed
+                if let pauseEndDate = pauseEndDate, Date() >= pauseEndDate {
+                    isPaused = false // Resume auto-play
+                }
+                return
+            }
+
+            withAnimation(.easeInOut(duration: Double(msTransitionTime) / 1000)) {
+                index += 1
+                if loop {
+                    expandDataIfNeeded()
+                    pruneDataIfNeeded()
+                } else {
+                    index = min(index, data.count - 1)
+                }
+            }
+        }
+    }
+
+    // MARK: - Offsets
+
+    private func xOffset(in totalWidth: CGFloat) -> CGFloat {
+        let itemWidth = cardWidth + spacing
+        let baseOffset = -CGFloat(index) * itemWidth
+        let centerAdjustment = (totalWidth - cardWidth) / 2
+        return baseOffset + translation + centerAdjustment
+    }
+
+    // MARK: - Drag Handling
+
+    private func handleDragEnd(translation: CGFloat) {
+        let threshold = cardWidth / 2
+
+        if translation < -threshold {
+            // Swipe left => next
+            index += 1
+        } else if translation > threshold {
+            // Swipe right => prev
+            index -= 1
+        }
+
+        if loop {
+            expandDataIfNeeded()
+            pruneDataIfNeeded()
+        } else {
+            // Non-loop clamp
+            index = max(0, min(index, data.count - 1))
+        }
+
+        // Pause auto-play for 10 seconds
+        pauseAutoPlay(for: 10)
```

Comment:
In android, I currently disable manual dragging if autoplay is enabled... I guess we want to allow this there? not sure if we might want it to be a setting though... (Not a blocker for this PR)